channels/rdpgfx/client/rdpgfx_main.c

```diff
@@ -689,19 +904,22 @@ static UINT rdpgfx_recv_cache_import_reply_pdu(RDPGFX_CHANNEL_CALLBACK* callback
 	if (!Stream_CheckAndLogRequiredLength(TAG, s, 2ull * pdu.importedEntriesCount))
 		return ERROR_INVALID_DATA;
 
-	pdu.cacheSlots = (UINT16*)calloc(pdu.importedEntriesCount, sizeof(UINT16));
+	if (pdu.importedEntriesCount > RDPGFX_CACHE_ENTRY_MAX_COUNT)
+		return ERROR_INVALID_DATA;
 
-	if (!pdu.cacheSlots)
+	if (Stream_GetRemainingLength(s) < (size_t)(pdu.importedEntriesCount * 2))
 	{
```

Comment:
`if (Stream_GetRemainingLength(s) / 2 < (size_t)pdu.importedEntriesCount)` would be better for int overflow